ang/civiawards/dashboard/directives/more-filters-dashboard-action-button.controller.js

```diff
@@ -0,0 +1,171 @@
+    var model = {
+      statuses: _.map(CaseStatus.getAll(), mapSelectOptions),
+      award_types: _.map(AwardType.getAll(), mapSelectOptions),
+      awardOptions: [
+        { text: ts('My Awards'), id: 'my_awards' },
+        { text: ts('All Awards'), id: 'all_awards' }
+      ],
+      selectedFilters: {
+        awardFilter: 'my_awards',
+        statuses: '',
+        award_types: '',
+        start_date: null,
+        end_date: null
+      },
+      applyFilterAndCloseDialog: applyFilterAndCloseDialog
+    };
+
+    $scope.clickHandler = clickHandler;
+    $scope.isVisible = isVisible;
+    $scope.isNotificationVisible = isNotificationVisible;
+
+    (function init () {
+      applyFilter();
+    }());
+
+    /**
+     * Checks if Notification should be visible
+     *
+     * @returns {boolean} if Notification should be visible
+     */
+    function isNotificationVisible () {
+      return !_.isEqual(model.selectedFilters, {
+        awardFilter: 'my_awards',
+        statuses: '',
+        award_types: '',
+        start_date: null,
+        end_date: null
+      });
+    }
+
+    /**
+     * Redirects the user to the awards creation screen.
+     */
+    function clickHandler () {
+      dialogService.open('MoreFilters', '~/civiawards/dashboard/directives/more-filters-popup.html', model, {
+        autoOpen: false,
+        height: 'auto',
+        width: '350px',
+        title: 'More Filters'
+      });
+    }
+
+    /**
+     *
+     */
+    function applyFilterAndCloseDialog () {
+      applyFilter();
+      dialogService.close('MoreFilters');
+    }
+    /**
+     *
+     */
+    function applyFilter () {
+      processMyAwardsFilter()
+        .then(processAwardTypeFilters)
+        .then(function (awardTypeIds) {
+          $rootScope.$broadcast('civicase::dashboard-filters::updated', {
+            case_type_id: { IN: awardTypeIds }
+          });
+        });
+    }
+
+    /**
+     * Process Award Type Filters
+     *
+     * @param {number[]} caseTypeIDs case type ids
+     * @returns {Promise} promise
+     */
+    function processAwardTypeFilters (caseTypeIDs) {
+      var filters = { sequential: 1 };
+
+      if (model.selectedFilters.award_types !== '') {
+        filters.award_type = { IN: getSelect2Value(model.selectedFilters.award_types) };
+      }
+      if (model.selectedFilters.start_date) {
+        filters.start_date = model.selectedFilters.start_date;
+      }
+      if (model.selectedFilters.end_date) {
+        filters.end_date = model.selectedFilters.end_date;
+      }
+      if (caseTypeIDs.length > 0) {
+        filters.case_type_id = { IN: caseTypeIDs };
+      }
+
+      return crmApi('AwardDetail', 'get', filters)
+        .then(function (awardsData) {
+          return awardsData.values.map(function (award) {
+            return award.case_type_id;
+          });
+        });
+    }
+
+    /**
+     * Process My Awards/All Awards filters
+     *
+     * @returns {Promise} promise
```

Comment:
should specify the resolve value. Can be `@returns {Promise<number[]>}` a promise that resolves to a list of award ids`.